package generator;

import java.awt.HeadlessException;
import javax.swing.ImageIcon;
import org.junit.*;
import static org.junit.Assert.*;
import com.badlogic.gdx.graphics.Texture;

/**
 * The class <code>DungeonTest</code> contains tests for the class <code>{@link Dungeon}</code>.
 *
 * @generatedBy CodePro at 11/06/19 12:57
 * @author Arthur Ferrão
 * @version $Revision: 1.0 $
 */
public class DungeonTest {
	/**
	 * Run the Dungeon() constructor test.
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testDungeon_1()
		throws Exception {
		Dungeon result = new Dungeon();
		assertNotNull(result);
		// add additional test code here
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_1()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_2()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_3()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_4()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_5()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_6()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_7()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_8()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_9()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_10()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the void fill(int[],int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testFill_11()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int[] pixels = new int[] {};
		int x = 1;
		int y = 1;
		int dim = 1;

		fixture.fill(pixels, x, y, dim);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		//       at generator.Dungeon.fill(Dungeon.java:600)
	}

	/**
	 * Run the ImageIcon generateDungeonImage() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testGenerateDungeonImage_1()
		throws Exception {
		Dungeon fixture = new Dungeon();

		ImageIcon result = fixture.generateDungeonImage();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.IllegalArgumentException: Width (0) and height (0) cannot be <= 0
		//       at java.awt.image.DirectColorModel.createCompatibleWritableRaster(Unknown Source)
		//       at java.awt.image.BufferedImage.<init>(Unknown Source)
		//       at generator.Dungeon.generateDungeonImage(Dungeon.java:582)
		assertNotNull(result);
	}

	/**
	 * Run the ImageIcon generateDungeonImage() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testGenerateDungeonImage_2()
		throws Exception {
		Dungeon fixture = new Dungeon();

		ImageIcon result = fixture.generateDungeonImage();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.IllegalArgumentException: Width (0) and height (0) cannot be <= 0
		//       at java.awt.image.DirectColorModel.createCompatibleWritableRaster(Unknown Source)
		//       at java.awt.image.BufferedImage.<init>(Unknown Source)
		//       at generator.Dungeon.generateDungeonImage(Dungeon.java:582)
		assertNotNull(result);
	}

	/**
	 * Run the ImageIcon generateDungeonImage() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testGenerateDungeonImage_3()
		throws Exception {
		Dungeon fixture = new Dungeon();

		ImageIcon result = fixture.generateDungeonImage();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.IllegalArgumentException: Width (0) and height (0) cannot be <= 0
		//       at java.awt.image.DirectColorModel.createCompatibleWritableRaster(Unknown Source)
		//       at java.awt.image.BufferedImage.<init>(Unknown Source)
		//       at generator.Dungeon.generateDungeonImage(Dungeon.java:582)
		assertNotNull(result);
	}

	/**
	 * Run the int getCell(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testGetCell_1()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int x = 1;
		int y = 1;

		int result = fixture.getCell(x, y);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.getCell(Dungeon.java:261)
		assertEquals(0, result);
	}

	/**
	 * Run the void main(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testMain_1()
		throws Exception {
		String[] args = new String[] {};

		Dungeon.main(args);

		// add additional test code here
	}

	/**
	 * Run the void main(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.awt.HeadlessException.class)
	public void testMain_2()
		throws Exception {
		String[] args = new String[] {};

		Dungeon.main(args);

		// add additional test code here
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_1()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 1;
		int cy = 1;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_2()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 0;
		int cy = 1;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_3()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 0;
		int cy = 0;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_4()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 0;
		int cy = 0;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_5()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 0;
		int cy = 0;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_6()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 1;
		int cy = 1;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the Texture peerGem(int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test(expected = java.lang.IllegalArgumentException.class)
	public void testPeerGem_7()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int cx = 1;
		int cy = 1;

		Texture result = fixture.peerGem(cx, cy);

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the void setCell(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testSetCell_1()
		throws Exception {
		Dungeon fixture = new Dungeon();
		int x = 1;
		int y = 1;
		int celltype = 1;

		fixture.setCell(x, y, celltype);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.ArrayIndexOutOfBoundsException: 1
		//       at generator.Dungeon.setCell(Dungeon.java:257)
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_1()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_2()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_3()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_4()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_5()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_6()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_7()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_8()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_9()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_10()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Run the String showDungeon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Test
	public void testShowDungeon_11()
		throws Exception {
		Dungeon fixture = new Dungeon();

		String result = fixture.showDungeon();

		// add additional test code here
		assertEquals("", result);
	}

	/**
	 * Perform pre-test initialization.
	 *
	 * @throws Exception
	 *         if the initialization fails for some reason
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@Before
	public void setUp()
		throws Exception {
		// add additional set up code here
	}

	/**
	 * Perform post-test clean-up.
	 *
	 * @throws Exception
	 *         if the clean-up fails for some reason
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	@After
	public void tearDown()
		throws Exception {
		// Add additional tear down code here
	}

	/**
	 * Launch the test.
	 *
	 * @param args the command line arguments
	 *
	 * @generatedBy CodePro at 11/06/19 12:57
	 */
	public static void main(String[] args) {
		new org.junit.runner.JUnitCore().run(DungeonTest.class);
	}
}